### PR TITLE
Migrate SubscriptionSettingsFragment to compose

### LINKED ISSUE #28: Migrate `SubscriptionSettingsFragment` to Jetpack Compose

### PR DESCRIPTION
**Pull request by sunkup**

### CODE REVIEW

File: app/src/main/java/at/bitfire/icsdroid/ui/SubscriptionSettingsFragment.kt [hunk 1]
```diff
@@ -5,157 +5,206 @@
...
-
         val colorPickerContract = registerForActivityResult(ColorPickerActivity.Contract()) { color ->
             model.color.value = color
         }
-        binding.color.setOnClickListener {
-            colorPickerContract.launch(model.color.value)
+        return ComposeView(requireActivity()).apply {
+            setContent {
+                SubscriptionSettingsComposable(
+                    url = model.url.observeAsState("").value,
```
Comment (ArnyminerZ) [inline app/src/main/java/at/bitfire/icsdroid/ui/SubscriptionSettingsFragment.kt]: I'd also use here the `by` operator. Ex:
```kotlin
val url by model.url.observeAsState("")
...
```

File: app/src/main/java/at/bitfire/icsdroid/ui/SubscriptionSettingsFragment.kt [hunk 2]
```diff
@@ -5,157 +5,206 @@
...
-        // We add the listener once the switch has an initial value
-        switch.setOnCheckedChangeListener(getOnCheckedChangeListener(switch, selectedMinutes))
-
-        if (min == null)
-            textView.text = getString(R.string.add_calendar_alarms_default_none)
-        else {
-            val alarmPeriodText = PeriodFormat.wordBased().print(Minutes.minutes(min.toInt()))
-            textView.text = getString(R.string.add_calendar_alarms_default_description, alarmPeriodText)
-        }
+    private fun stringToLongOrNull(input: String) = try {
```
Comment (ArnyminerZ) [inline app/src/main/java/at/bitfire/icsdroid/ui/SubscriptionSettingsFragment.kt]: Why not using [`toLongOrNull`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/to-long-or-null.html)? 🤔

**Review by ArnyminerZ — CHANGES_REQUESTED** [review]

File: app/src/main/java/at/bitfire/icsdroid/ui/SubscriptionSettingsFragment.kt (hunk 2 above)
Comment (sunkup) [reply app/src/main/java/at/bitfire/icsdroid/ui/SubscriptionSettingsFragment.kt]: Oh, true haha.

File: app/src/main/java/at/bitfire/icsdroid/ui/SubscriptionSettingsFragment.kt (hunk 1 above)
Comment (sunkup) [reply app/src/main/java/at/bitfire/icsdroid/ui/SubscriptionSettingsFragment.kt]: yup, kinda irritating, but better be consistent

File: app/src/main/java/at/bitfire/icsdroid/ui/EditCalendarActivity.kt [hunk 3]
```diff
@@ -246,7 +242,7 @@ class EditCalendarActivity: AppCompatActivity() {
          */
         fun updateSubscription(
             subscriptionSettingsModel: SubscriptionSettingsFragment.SubscriptionSettingsModel,
-            credentialsModel: CredentialsModel
+            credentialsModel: CredentialsFragment.CredentialsModel
```
Comment (ArnyminerZ) [inline app/src/main/java/at/bitfire/icsdroid/ui/EditCalendarActivity.kt]: I think this should be just `CredentialsModel`

File: app/src/main/java/at/bitfire/icsdroid/ui/SubscriptionSettingsFragment.kt [hunk 4]
```diff
@@ -5,157 +5,208 @@
...
+                val defaultAllDayAlarmMinutes by model.defaultAllDayAlarmMinutes.observeAsState()
+                SubscriptionSettingsComposable(
+                    url = url,
+                    title = title,
+                    titleChanged = { model.title.postValue(it) },
+                    color = color,
+                    colorIconClicked = { colorPickerContract.launch(color) },
+                    ignoreAlerts = ignoreAlerts,
+                    ignoreAlertsChanged = { model.ignoreAlerts.postValue(it) },
+                    defaultAlarmMinutes = (defaultAlarmMinutes ?: "").toString(),
```
Comment (ArnyminerZ) [inline app/src/main/java/at/bitfire/icsdroid/ui/SubscriptionSettingsFragment.kt]: Can't we just accept here `Int?`?

File: app/src/main/java/at/bitfire/icsdroid/ui/SubscriptionSettingsFragment.kt [hunk 5]
```diff
@@ -5,157 +5,208 @@
...
+                    defaultAlarmMinutesChanged = { model.defaultAlarmMinutes.postValue(it.toLongOrNull()) },
+                    defaultAllDayAlarmMinutes = (defaultAllDayAlarmMinutes ?: "").toString(),
```
Comment (ArnyminerZ) [inline app/src/main/java/at/bitfire/icsdroid/ui/SubscriptionSettingsFragment.kt]: Same here, doing the conversion inside of the Composable.

**Review by ArnyminerZ — CHANGES_REQUESTED** [review]

File: app/src/main/java/at/bitfire/icsdroid/ui/SubscriptionSettingsFragment.kt (hunk 4 above)
Comment (sunkup) [reply app/src/main/java/at/bitfire/icsdroid/ui/SubscriptionSettingsFragment.kt]: yup, and if I remember correctly that's even best practice :+1:

File: app/src/main/java/at/bitfire/icsdroid/ui/SubscriptionSettingsFragment.kt (hunk 5 above)
Comment (sunkup) [reply app/src/main/java/at/bitfire/icsdroid/ui/SubscriptionSettingsFragment.kt]: yup, and if I remember correctly that's even best practice :+1:

File: app/src/main/java/at/bitfire/icsdroid/ui/EditCalendarActivity.kt (hunk 3 above)
Comment (sunkup) [reply app/src/main/java/at/bitfire/icsdroid/ui/EditCalendarActivity.kt]: Ah yes, overlooked that one

**Review by ArnyminerZ — APPROVED** [review]

**Pull request merged**